docker/src/main/java/org/arquillian/cube/impl/util/BindingUtil.java

```diff
@@ -20,12 +20,21 @@ private BindingUtil() {
     }
 
     public static Binding binding(DockerClientExecutor executor, String cubeId) {
-        InspectContainerResponse inspectResponse = executor.getDockerClient().inspectContainerCmd(cubeId).exec();
+        InspectContainerResponse inspectResponse = executor.getDockerClient().inspectContainerCmd( cubeId ).exec();
 
         HostConfig hostConfig = inspectResponse.getHostConfig();
+
+        /**
+         * This isn't actually the gateway IP of the host, it's the ip of the gateway within the docker runtime,
+         * which may not be accessible to us.  We need the IP the client uses, that will be routable, and will
+         * allow us to validate the ports in later phases
         String gatewayIp = inspectResponse.getNetworkSettings().getGateway();
 
-        Binding binding = new Binding(gatewayIp);
+         **/
+
+        final String dockerIp = executor.getDockerUri().getHost();
+        Binding binding = new Binding(dockerIp);
```

Comment:
Sorry to my question but I have a headache with ip and ports in docker. I mean originally we used docker server ip and we had some problems with polling strategy because it returned true immediately not when the container port was ready.